executor/common_kvm_arm64_syzos.h

```diff
@@ -11,10 +11,69 @@
 // Start/end of the guest section.
 extern char *__start_guest, *__stop_guest;
 
+#define dmb() asm volatile("dmb sy" \
+			   :        \
+			   :        \
+			   : "memory")
+
+#define SYZOS_CALL5(id, arg1, arg2, arg3, arg4, arg5)                                                                         \
+	do {                                                                                                                  \
+		dmb();                                                                                                        \
+		uint64 fn = ((uint64*)ARM64_ADDR_API_TRAMPOLINES)[id];                                                        \
+		dmb();                                                                                                        \
+		void (*api)(uint64, uint64, uint64, uint64, uint64) = (void (*)(uint64, uint64, uint64, uint64, uint64))(fn); \
+		(api)(arg1, arg2, arg3, arg4, arg5);                                                                          \
+	} while (0)
+
+typedef enum {
```

Comment:
We have this enum + all values need to be duplicated and mapped to functions in syzos_api_functions + fill_api_trampolines.

I am thinking if this can be done simpler if we have a switch inside of interpret_payload that will do everything.
Something along the lines of:

```
switch (api) {
case 0: return guest_ucall(args);
case 1: return some_other_api_call(args);
...
}
```
    
Then we don't need any additional enums/mappings/trampolines.

I am not sure we can do direct calls here. Can we? It would nice b/c if we will have complex api calls, we will need helper functions.
But if not, then I think we can do with some macro that will convert the address:
```
case 0: return CALL(guest_ucall(args));
```

Reply:
Yes, I think it's better to pass generated code chunk to syzos, rather then complicate and extend ifuzz more.
If we go syzos way, then it's better to concentrate all new smartness in syzos, and stop extending ifuzz (keep it just as random instruction generation engine, maybe even remove pseudo-instruction support at some point since it's easier/better to do them in syzos).

Reply:
Ok, then we don't need trampolines then (I removed them already)